src/System.Net.Security/src/System/Net/SecurityStatusAdapterPal.Windows.cs

```diff
@@ -0,0 +1,82 @@
+            { Interop.SecurityStatus.InvalidHandle, SecurityStatusPalErrorCode.InvalidHandle },
+            { Interop.SecurityStatus.InvalidToken, SecurityStatusPalErrorCode.InvalidToken },
+            { Interop.SecurityStatus.LogonDenied, SecurityStatusPalErrorCode.LogonDenied },
+            { Interop.SecurityStatus.NoAuthenticatingAuthority, SecurityStatusPalErrorCode.NoAuthenticatingAuthority },
+            { Interop.SecurityStatus.NoImpersonation, SecurityStatusPalErrorCode.NoImpersonation },
+            { Interop.SecurityStatus.NoCredentials, SecurityStatusPalErrorCode.NoCredentials },
+            { Interop.SecurityStatus.NotOwner, SecurityStatusPalErrorCode.NotOwner },
+            { Interop.SecurityStatus.OK, SecurityStatusPalErrorCode.OK },
+            { Interop.SecurityStatus.OutOfMemory, SecurityStatusPalErrorCode.OutOfMemory },
+            { Interop.SecurityStatus.OutOfSequence, SecurityStatusPalErrorCode.OutOfSequence },
+            { Interop.SecurityStatus.PackageNotFound, SecurityStatusPalErrorCode.PackageNotFound },
+            { Interop.SecurityStatus.MessageAltered, SecurityStatusPalErrorCode.MessageAltered },
+            { Interop.SecurityStatus.QopNotSupported, SecurityStatusPalErrorCode.QopNotSupported },
+            { Interop.SecurityStatus.Renegotiate, SecurityStatusPalErrorCode.Renegotiate },
+            { Interop.SecurityStatus.SecurityQosFailed, SecurityStatusPalErrorCode.SecurityQosFailed },
+            { Interop.SecurityStatus.SmartcardLogonRequired, SecurityStatusPalErrorCode.SmartcardLogonRequired },
+            { Interop.SecurityStatus.TargetUnknown, SecurityStatusPalErrorCode.TargetUnknown },
+            { Interop.SecurityStatus.TimeSkew, SecurityStatusPalErrorCode.TimeSkew },
+            { Interop.SecurityStatus.UnknownCredentials, SecurityStatusPalErrorCode.UnknownCredentials },
+            { Interop.SecurityStatus.UnsupportedPreauth, SecurityStatusPalErrorCode.UnsupportedPreauth },
+            { Interop.SecurityStatus.Unsupported, SecurityStatusPalErrorCode.Unsupported },
+            { Interop.SecurityStatus.UntrustedRoot, SecurityStatusPalErrorCode.UntrustedRoot },
+            { Interop.SecurityStatus.WrongPrincipal, SecurityStatusPalErrorCode.WrongPrincipal }
+        };
+
+        internal static SecurityStatusPal GetSecurityStatusPalFromNativeInt(int win32SecurityStatus)
+        {
+            return GetSecurityStatusPalFromInterop((Interop.SecurityStatus) win32SecurityStatus);
+        }
+
+        internal static SecurityStatusPal GetSecurityStatusPalFromInterop(Interop.SecurityStatus win32SecurityStatus)
+        {
+            SecurityStatusPalErrorCode statusCode;
+
+            if (!s_statusDictionary.TryGetForward(win32SecurityStatus, out statusCode))
+            {
+                Debug.Fail("Unknown Interop.SecurityStatus value: " + win32SecurityStatus);
+                throw new InternalException();
```

Comment:
Is throwing here and below really the right behavior?  We'll want to clean this up post-merge with https://github.com/dotnet/corefx/issues/5890, but I'm wondering if this is the right behavior for RC2.